List of NETSDK error messages (#29987)
dotnet/docs: docs/core/tools/sdk-errors/index.md

---
title: ".NET SDK error list"
description: A complete list of NETSDKxxxx errors, with links to more info where more info is available.
ms.topic: error-reference
ms.date: 06/23/2022
f1_keywords:
- NETSDK
---
# .NET SDK error list

**This article applies to:** ✔️ .NET 6 SDK and later versions

This is a complete list of the errors that you might get from the .NET SDK while developing .NET apps. If more info is available for a particular error, the error number is a link.

| SDK Message Number | Message            |
|--------------------|--------------------|
|NETSDK1001|At least one possible target framework must be specified.|

> |NETSDK1002|Project '{0}' targets '{2}'. It cannot be referenced by a project that targets '{1}'.|

Author: These were all extracted verbatim from the SDK source code. We can change parameter numbers here but I'm not sure it would be worth the effort.

|NETSDK1007|Cannot find project info for '{0}'. This can indicate a missing project reference.|
|NETSDK1008|Missing '{0}' metadata on '{1}' item '{2}'.|
|NETSDK1009|Unrecognized preprocessor token '{0}' in '{1}'.|
|NETSDK1010|The '{0}' task must be given a value for parameter '{1}' in order to consume preprocessed content.|
|NETSDK1011|Assets are consumed from project '{0}', but no corresponding MSBuild project path was  found in '{1}'.|
|NETSDK1012|Unexpected file type for '{0}'. Type is both '{1}' and '{2}'.|
|[NETSDK1013](netsdk1013.md)|The TargetFramework value '{0}' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly.|
|NETSDK1014|Content item for '{0}' sets '{1}', but does not provide  '{2}' or '{3}'.|
|NETSDK1015|The preprocessor token '{0}' has been given more than one value. Choosing '{1}' as the value.|
|NETSDK1016|Unable to find resolved path for '{0}'.|
|NETSDK1017|Asset preprocessor must be configured before assets are processed.|
|NETSDK1018|Invalid NuGet version string: '{0}'.|
|NETSDK1019|{0} is an unsupported framework.|

> |NETSDK1020|Package Root {0} was incorrectly given for Resolved library {1}|

Author: We could but here too I'm not sure it would be worth the time spent on it.

|NETSDK1021|More than one file found for {0}|

> |[NETSDK1022](netsdk1022.md)|Duplicate '{0}' items were included. The .NET SDK includes '{0}' items from your project directory by default. You can either remove these items from your project file, or set the '{1}' property to '{2}' if you want to explicitly include them in your project file. For more information, see {4}. The duplicate items were: {3}|

Author: We could change instances like this but I'm not sure it's worth the time to do so.

|NETSDK1046|The TargetFramework value '{0}' is not valid. To multi-target, use the 'TargetFrameworks' property instead.|
|NETSDK1047|Assets file '{0}' doesn't have a target for '{1}'. Ensure that restore has run and that you have included '{2}' in the TargetFrameworks for your project. You may also need to include '{3}' in your project's RuntimeIdentifiers.|
|NETSDK1048|'AdditionalProbingPaths' were specified for GenerateRuntimeConfigurationFiles, but are being skipped because 'RuntimeConfigDevPath' is empty.|
|NETSDK1049|Resolved file has a bad image, no metadata, or is otherwise inaccessible. {0} {1}|
|NETSDK1050|The version of Microsoft.NET.Sdk used by this project is insufficient to support references to libraries targeting .NET Standard 1.5 or higher.  Please install version 2.0 or higher of the .NET Core SDK.|
|NETSDK1051|Error parsing FrameworkList from '{0}'.  {1} '{2}' was invalid.|
|NETSDK1052|Framework list file path '{0}' is not rooted. Only full paths are supported.|
|NETSDK1053|Pack as tool does not support self contained.|

> |NETSDK1054|only supports .NET Core.|

Reviewer: ```suggestion
|NETSDK1054|Only supports .NET Core.|
```

|NETSDK1055|DotnetTool does not support target framework lower than netcoreapp2.1.|
|NETSDK1056|Project is targeting runtime '{0}' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.|

> |NETSDK1057|You are using a preview version of .NET. See: aka.ms/dotnet-support-policy|

Reviewer: ```suggestion
|NETSDK1057|You are using a preview version of .NET. See <https://aka.ms/dotnet-support-policy>.|
```

Author: I had removed https:// because markdownlint complained about bare URLs.  I'll try this and see if it elicits the same response.

|NETSDK1058|Invalid value for ItemSpecToUse parameter: '{0}'.  This property must be blank or set to 'Left' or 'Right'|

> |[NETSDK1059](netsdk1059.md)|The tool '{0}' is now included in the .NET SDK. Information on resolving this warning is available at (aka.ms/dotnetclitools-in-box).|

Reviewer: ```suggestion
|[NETSDK1059](netsdk1059.md)|The tool '{0}' is now included in the .NET SDK. Information on resolving this warning is available at <https://aka.ms/dotnetclitools-in-box>.|
```

|NETSDK1063|The path to the project assets file was not set. Run a NuGet package restore to generate this file.|
|[NETSDK1064](netsdk1064.md)|Package {0}, version {1} was not found. It might have been deleted since NuGet restore. Otherwise, NuGet restore might have only partially completed, which might have been due to maximum path length restrictions.|
|NETSDK1065|Cannot find app host for {0}. {0} could be an invalid runtime identifier (RID). For more information about RID, see aka.ms/rid-catalog.|
|NETSDK1067|Self-contained applications are required to use the application host. Either set SelfContained to false or set UseAppHost to true.|
|NETSDK1068|The framework-dependent application host requires a target framework of at least 'netcoreapp2.1'.|
|NETSDK1069|This project uses a library that targets .NET Standard 1.5 or higher, and the project targets a version of .NET Framework that doesn't have built-in support for that version of .NET Standard. Visit aka.ms/net-standard-known-issues for a set of known issues. Consider retargeting to .NET Framework 4.7.2.|
|NETSDK1070|The application configuration file must have root configuration element.|

> |[NETSDK1071](netsdk1071.md)|A PackageReference to '{0}' specified a Version of `{1}`. Specifying the version of this package is not recommended. For more information, see aka.ms/sdkimplicitrefs|

Reviewer: ```suggestion
|[NETSDK1071](netsdk1071.md)|A PackageReference to '{0}' specified a Version of `{1}`. Specifying the version of this package is not recommended. For more information, see <https://aka.ms/sdkimplicitrefs>.|
```

|NETSDK1077|Failed to lock resource.|
|NETSDK1078|Unable to use '{0}' as application host executable because it's not a Windows PE file.|
|[NETSDK1079](netsdk1079.md)|The Microsoft.AspNetCore.All package is not supported when targeting .NET Core 3.0 or higher.  A FrameworkReference to Microsoft.AspNetCore.App should be used instead, and will be implicitly included by Microsoft.NET.Sdk.Web.|
|NETSDK1080|A PackageReference to Microsoft.AspNetCore.App is not necessary when targeting .NET Core 3.0 or higher. If Microsoft.NET.Sdk.Web is used, the shared framework will be referenced automatically. Otherwise, the PackageReference should be replaced with a FrameworkReference.|
|NETSDK1081|The targeting pack for {0} was not found. You may be able to resolve this by running a NuGet restore on the project.|
|NETSDK1082|There was no runtime pack for {0} available for the specified RuntimeIdentifier '{1}'.|
|NETSDK1083|The specified RuntimeIdentifier '{0}' is not recognized.|
|NETSDK1084|There is no application host available for the specified RuntimeIdentifier '{0}'.|
|NETSDK1085|The 'NoBuild' property was set to true but the 'Build' target was invoked.|

> |NETSDK1086|A FrameworkReference for '{0}' was included in the project. This is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see {1}|

Author: We would have to find the code that generates this error and see what it uses for parameter 1.

|NETSDK1122|ReadyToRun compilation will be skipped because it is only supported for .NET Core 3.0 or higher.|
|NETSDK1123|Publishing an application to a single-file requires .NET Core 3.0 or higher.|
|NETSDK1124|Trimming assemblies requires .NET Core 3.0 or higher.|
|NETSDK1125|Publishing to a single-file is only supported for netcoreapp target.|
|NETSDK1126|Publishing ReadyToRun using Crossgen2 is only supported for self-contained applications.|
|NETSDK1127|The targeting pack {0} is not installed. Please restore and try again.|
|NETSDK1128|COM hosting does not support self-contained deployments.|
|NETSDK1129|The 'Publish' target is not supported without specifying a target framework. The current project targets multiple frameworks, you must specify the framework for the published application.|
|[NETSDK1130](netsdk1130.md)|{1} cannot be referenced. Referencing a Windows Metadata component directly when targeting .NET 5 or higher is not supported. For more information, see aka.ms/netsdk1130|
|NETSDK1131|Producing a managed Windows Metadata component with WinMDExp is not supported when targeting {0}.|
|NETSDK1132|No runtime pack information was available for {0}.|

> |NETSDK1133|There was conflicting information about runtime packs available for {0}:|

Reviewer: ```suggestion
|NETSDK1133|There was conflicting information about runtime packs available for {0}.|
```

|NETSDK1134|Building a solution with a specific RuntimeIdentifier is not supported. If you would like to publish for a single RID, specifiy the RID at the individual project level instead.|
|NETSDK1135|SupportedOSPlatformVersion {0} cannot be higher than TargetPlatformVersion {1}.|
|NETSDK1136|The target platform must be set to Windows (usually by including '-windows' in the TargetFramework property) when using Windows Forms or WPF, or referencing projects or packages that do so.|
|NETSDK1137|It is no longer necessary to use the Microsoft.NET.Sdk.WindowsDesktop SDK. Consider changing the Sdk attribute of the root Project element to 'Microsoft.NET.Sdk'.|

> |NETSDK1138|The target framework '{0}' is out of support and will not receive security updates in the future. Please refer to {1} for more information about the support policy.|

Author: We'd have to look up in the source code what it plugs in here.

|NETSDK1139|The target platform identifier {0} was not recognized.|

> |NETSDK1140|{0} is not a valid TargetPlatformVersion for {1}. Valid versions include:|

Author: I expect this could vary substantially depending on what {0} is.

|NETSDK1146|PackAsTool does not support TargetPlatformIdentifier being set. For example, TargetFramework cannot be net5.0-windows, only net5.0. PackAsTool also does not support UseWPF or UseWindowsForms when targeting .NET 5 and higher.|
|[NETSDK1147](netsdk1147.md)|To build this project, the following workloads must be installed: {0}|
|NETSDK1148|A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll. Please update to a newer .NET SDK in order to reference this assembly.|
|[NETSDK1149](netsdk1149.md)|{0} cannot be referenced because it uses built-in support for WinRT, which is no longer supported in .NET 5 and higher.  An updated version of the component supporting .NET 5 is needed. For more information, see aka.ms/netsdk1149|
|NETSDK1150|The referenced project '{0}' is a non self-contained executable.  A non self-contained executable cannot be referenced by a self-contained executable.  For more information, see aka.ms/netsdk1150|

> |NETSDK1151|The referenced project '{0}' is a self-contained executable.  A self-contained executable cannot be referenced by a non self-contained executable.  For more information, see aka.ms/netsdk1151|

Reviewer: ```suggestion
|NETSDK1151|The referenced project '{0}' is a self-contained executable.  A self-contained executable cannot be referenced by a non self-contained executable.  For more information, see <https://aka.ms/netsdk1151>.|
```

|NETSDK1152|Found multiple publish output files with the same relative path: {0}.|
|NETSDK1153|CrossgenTool not specified in PDB compilation mode.|
|NETSDK1154|Crossgen2Tool must be specified when UseCrossgen2 is set to true.|
|NETSDK1155|Crossgen2Tool executable '{0}' not found.|
|NETSDK1156|.NET host executable '{0}' not found.|
|NETSDK1157|JIT library '{0}' not found.|
|NETSDK1158|Required '{0}' metadata missing on Crossgen2Tool item.|
|NETSDK1159|CrossgenTool must be specified when UseCrossgen2 is set to false.|
|NETSDK1160|CrossgenTool executable '{0}' not found.|
|NETSDK1161|DiaSymReader library '{0}' not found.|
|NETSDK1162|PDB generation: R2R executable '{0}' not found.|
|NETSDK1163|Input assembly '{0}' not found.|
|NETSDK1164|Missing output PDB path in PDB generation mode (OutputPDBImage metadata).|
|NETSDK1165|Missing output R2R image path (OutputR2RImage metadata).|
|NETSDK1166|Cannot emit symbols when publishing for .NET 5 with Crossgen2 using composite mode.|

> |NETSDK1167|Compression in a single file bundle is only supported when publishing for .NET6 or higher.|

Reviewer: ```suggestion
|NETSDK1167|Compression in a single file bundle is only supported when publishing for .NET 6 or higher.|
```

> |NETSDK1168|WPF is not supported or recommended with trimming enabled. Please go to aka.ms/dotnet-illink/wpf for more details.|

Reviewer: ```suggestion
|NETSDK1168|WPF is not supported or recommended with trimming enabled. Please go to <https://aka.ms/dotnet-illink/wpf> for more details.|
```

|NETSDK1169|The same resource ID {0} was specified for two type libraries '{1}' and '{2}'. Duplicate type library IDs are not allowed.|
|NETSDK1170|The provided type library ID '{0}' for type libary '{1}' is invalid. The ID must be a positive integer less than 65536.|
|NETSDK1171|An integer ID less than 65536 must be provided for type library '{0}' because more than one type library is specified.|
|NETSDK1172|The provided type library '{0}' does not exist.|
|NETSDK1173|The provided type library '{0}' is in an invalid format.|
|[NETSDK1174](netsdk1174.md)|The abbreviation of -p for --project is deprecated. Please use --project.|

> |NETSDK1175|Windows Forms is not supported or recommended with trimming enabled. Please go to aka.ms/dotnet-illink/windows-forms for more details.|

Reviewer: ```suggestion
|NETSDK1175|Windows Forms is not supported or recommended with trimming enabled. Please go to <https://aka.ms/dotnet-illink/windows-forms> for more details.|
```
